src/managers/PlayerManager.ts

```diff
@@ -46,6 +47,36 @@ class PlayerManager {
         );
     }
 
+    public getRoles(): string[] {
+        const seed: string = GeneralManager.getSeed();
+        const allRoles: string[] = this.generateRoles();
+        const shuffledRoles: string[] = seededShuffle(allRoles, seed);
+        return shuffledRoles;
+    }
+
+    public generateRoles(): string[] {
```

Comment:
In the rules, the available roles are always 1 more than the amount of players. For example, with 3 players there is 1 Saboteur and 3 gold miners. I think it's a lot easier the way you implemented it with same amount of roles as players. Well done!

Reply:
Really dont understand how this should work, if there is one more role available than amount of players, it is possible that there is no saboteur in the game, what's the purpose of having one more role?
Im fine with the way it is right now

Reply:
Its from the original rules. It gives uncertainty on how many saboteurs there actually are in every round. We could implement this too, but is not high priority.